DATM/AtmForce.F90

```diff
@@ -91,31 +89,51 @@ subroutine AtmForce(gcomp,exportState,externalClock,initmode,rc)
     call ESMF_FieldBundleGet(AtmBundleFwd, &
                              fieldName=trim(AtmBundleFields(ii)%field_name)//'_fwd', &
                              field=field, rc=rc)
-    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
-      line=__LINE__, &
-      file=__FILE__)) &
-      return  ! bail out
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
 
     call ESMF_FieldRead(field, &
                         fileName=trim(filename), &
                         variableName = trim(varname), &
                         rc=rc)
-    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
-      line=__LINE__, &
-      file=__FILE__)) &
-      return  ! bail out
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
 
     call ESMF_FieldGet(field,farrayPtr=AtmBundleFields(ii)%farrayPtr_fwd,rc=rc)
-    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
-      line=__LINE__, &
-      file=__FILE__)) &
-      return  ! bail out
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
 
      write(msgString,'(i6,2a18,f14.5)')ii,' inside AtmForce  ',trim(varname), &
                              AtmBundleFields(ii)%farrayPtr_fwd(iprnt,jprnt)
-     call ESMF_LogWrite(trim(msgString), ESMF_LOGMSG_INFO, rc=rc)
+     call ESMF_LogWrite(trim(msgString), ESMF_LOGMSG_INFO)
    endif !isPresent
   enddo
 
+  ! Check for fields which are not Present but are needed
+  ! Not very clean---would have been better to create this field in the forcing file
+  nfields = size(AtmBundleFields)
+  do ii = 1,nfields
+   if(.not.AtmBundleFields(ii)%isPresent)then
+    varname = trim(AtmBundleFields(ii)%standard_name)
+
+    ! get the '_fwd' field
+    call ESMF_FieldBundleGet(AtmBundleFwd, &
+                             fieldName=trim(AtmBundleFields(ii)%field_name)//'_fwd', &
+                             field=field, rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    call ESMF_FieldGet(field,farrayPtr=AtmBundleFields(ii)%farrayPtr_fwd,rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    if(trim(varname) .eq. 'mean_net_lw_flx')then
+         iid = 0; iiu = 0
+      do iii = 1,nfields
+       if(trim(AtmBundleFields(iii)%standard_name) == 'mean_down_lw_flx')iid = iii
+       if(trim(AtmBundleFields(iii)%standard_name) ==   'mean_up_lw_flx')iiu = iii
+      enddo
+
```

Comment:
maybe add here: if(iid>0 .and. iiu>0) then

Reply:
Yes, I will add that and then merge the PR.